dlrover/python/elastic_agent/master_client.py

```diff
@@ -194,12 +194,19 @@ def report_shard_checkpoint(self, shard_checkpoint):
         return self._stub.report_shard_checkpoint(request)
 
     @retry_grpc_request
-    def report_used_resource(self, memory, cpu):
+    def report_used_resource(self, memory, cpu, gpu_stats):
         request = elastic_training_pb2.ReportUsedResourceRequest()
         request.memory = memory
         request.cpu = cpu
+        for gpu in gpu_stats:
+            gpu_stats_message = request.gpu_stats[gpu["index"]]
```

Comment:
gpu_stats should be `list[GPUMetric]` and `gpu["index"]` should be `gpu.index`?

Reply:
Yes. Rewrited this with GPUMetric.